modules/ml/src/main/java/org/apache/ignite/ml/util/Utils.java

```diff
@@ -130,4 +132,50 @@
                 Spliterators.spliteratorUnknownSize(iter, Spliterator.ORDERED),
                 false);
     }
+
+    /**
+     * Zips two streams (in functional sense of zipping) i.e. returns stream consisting
+     * of results of applying zipper to corresponding entries of two stream.
+     *
+     * @param a First stream.
+     * @param b Second stream.
+     * @param zipper Bi-function combining two streams.
+     * @param <A> Type of first stream entries.
+     * @param <B> Type of secong stream entries.
+     * @param <C> Type of zipper output.
+     * @return Two streams zipped together.
+     */
+    public static<A, B, C> Stream<C> zip(Stream<? extends A> a,
```

Comment:
You don't use zip in code
Remove it please

Reply:
Removed.